elements/bulbs-reading-list/components/bulbs-reading-list-item.js

```diff
@@ -3,14 +3,146 @@ import {
+    invariant(this.dataset.contentAnalyticsDimensions,
+      markdownText + 'data-content-analytics-dimensions');
+
+    InViewMonitor.add(this);
+    this.id = parseInt(this.dataset.id, 10);
+    this.href = this.dataset.href;
+    this.partialUrl = this.dataset.partialUrl;
+    this.title = this.dataset.title;
+    this.gaDimensions = this.getGaDimensions();
+
+    this.isLoaded = false;
+    this.fetchPending = false;
+    this.loadingTemplate = '<p><i class="fa fa-spinner fa-spin"></i> Loading...</p>';
+
+    this.registerEvents();
+  }
+
+  registerEvents () {
+    this.addEventListener('approachingviewport', this.loadContent.bind(this));
+    this.addEventListener('pagestart', this.handlePageStart.bind(this));
+  }
+
+  getGaDimensions () {
+    let targeting = JSON.parse(
+      this.dataset.contentAnalyticsDimensions
+    );
+    return {
+      'dimension1': targeting.dimension1 || 'None',
+      'dimension2': targeting.dimension2 || 'None',
+      'dimension3': targeting.dimension3 || 'None',
+      'dimension4': targeting.dimension4 || 'None',
+      'dimension5': targeting.dimension5 || 'None',
+      'dimension6': targeting.dimension6 || 'None',
+      'dimension7': targeting.dimension7 || 'None',
+      'dimension8': targeting.dimension8 || 'None',
+      'dimension9': targeting.dimension9 || 'None',
+      'dimension10': targeting.dimension10 || 'None',
+      'dimension11': targeting.dimension11 || 'None',
+    };
+  }
+
+  prepGaTracker () {
+    return prepGaEventTracker(
+      'pageview',
+      window.GOOGLE_ANALYTICS_ID,
+      this.dimensions,
+    );
+  }
+
+  sendAnalyticsEvent () {
+    this.gaTrackerWrapper(
+      'send', 'event',
+      'reading_list',
+      'scroll_view',
+      this.href,
+    );
+  }
+
+  loadContent () {
+    return new Promise((resolve, reject) => {
+      if (this.shouldLoad()) {
+        this.fetchPending = true;
+        this.fillContent(this.loadingTemplate);
+        fetch(this.partialUrl, {
+          credentials: 'same-origin',
+        })
+          .then(filterBadResponse)
+          .then(getResponseText)
+          .then((response) => {
+            this.handleLoadContentComplete(response);
+            resolve(this);
+          })
+          .catch((response) => {
+            this.handleLoadContentError(response);
+            reject(this);
+          });
+      }
+    });
+  }
+
+  shouldLoad () {
+    return !(this.isLoaded || this.fetchPending);
+  }
+
+  fillContent (content) {
+    this.innerHTML = content;
+    this.dataset.loadStatus = 'loading';
+  }
+
+  handleLoadContentComplete (content) {
+    this.fillContent(content);
+    this.isLoaded = true;
+    this.fetchPending = false;
+    this.dataset.loadStatus = 'loaded';
   }
+
+  handleLoadContentError (response) {
+    this.fetchPending = false;
+    return new Promise((resolve, reject) => reject(`ReadingListArticle.loadContent(): fetch failed "${response.status} ${response.statusText}"`));
```

Comment:
Should update this to reflect that it's now `BulbsReadingListItem` instead of `ReadingListArticle`